### PR TITLE
Use configuration index file and mount USB drive

### LINKED ISSUE #1143: Labware: Persistent Labware Definitions and Calibrations
As a user, I would like my labware definitions and calibrated offsets to persist, even if I update my api or docker container (OT does a Resin update / moves my robot to a different app)

## Acceptance Criteria
- Move labware definitions out of Docker, probably into the USB thumb drive

## Implementation Notes
- How do we unpack the zip packet
- Need an overwrite flag

### PR DESCRIPTION
**Pull request by btmorr**
## overview

Auto-mount USB drive if present. Detect configuration index file, and create one if it does not exist. Use configuration index file to find all other settings files, instead of environment variables. Fixes #1143 (more followup may be desired, but this is sufficient for now)

## changelog

- (feature) Auto-mount USB drive if present
- (refactor) Use config index file instead of environment variables
- (feature) Auto-generate index file if it does not exist

## review requests

Tested on 🌔 🌔 both with and without USB drive present

This refactor changes several things about the way we manage app configuration:
- Reduces dependence on environment variables that are prone to mis-management by Resin
- Reduces the use of `opentrons.util.environment` (which abstracts over environment variables in a way that makes the environment hard to interpret) and creates an alternative pattern for configuration so that `opentrons.util.environment` can be fully removed in a future PR
- If the USB drive is available (which it should be on all robots, but auto-mounting requires a Dockerfile from this commit or later), moves configuration data into the USB drive so that robots can be moved between Resin apps without having their persistent data deleted
- Manages test data through a testing index instead of mocks/monkeypatch

### CODE REVIEW

File: api/Makefile [hunk 1]
```diff
@@ -105,4 +105,4 @@ change-left:
 # TODO(mc, 2018-03-14): use $(host) var
 .PHONY: term
 term:
-	ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@fd00:0:cafe:fefe::1
+	ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no root@$(host)
```
Comment (mcous) [inline api/Makefile]: Default $(host) has square brackets. Does this still work?

Comment (btmorr) [reply api/Makefile]: Yes. Checked.

File: api/opentrons/config/__init__.py [hunk 2]
```diff
@@ -1,3 +1,227 @@
...
+        res = resin_settings_dir
+    else:
+        new_path, new_cfg = _generate_base_config()
+        write_base_config(new_path, new_cfg)
+        res = new_path
+
+    return res
+
+
+def get_config_index() -> dict:
```
Comment (mcous) [inline api/opentrons/config/__init__.py]: I couldn't ever find a usage of this function that wasn't immediately proceeded by a `get`. Given that, would a `def get_config_file(module)` function would be a good idea?

File: api/opentrons/server/main.py [hunk 3]
```diff
@@ -123,7 +123,7 @@ def init(loop=None):
     server.app.router.add_post(
         '/server/update', update.install_api)
     server.app.router.add_post(
-        '/server/update_firmware', update.update_firmware)
+        '/server/update/firmware', update.update_firmware)
```
Comment (mcous) [inline api/opentrons/server/main.py]: 👍 on this change (so please don't revert) but kind of unexpected to see it tossed in to this PR

**Pull request merged**